src/state/slices/opportunitiesSlice/resolvers/thorchainLending/utils.ts

```diff
@@ -0,0 +1,153 @@
+
+  try {
+    const url =
+      `${REACT_APP_THORCHAIN_NODE_URL}/lcd/thorchain/quote/loan/open` +
+      `?from_asset=${collateralAssetId}` +
+      `&amount=${amountCryptoThorBaseUnit.toString()}` +
+      `&to_asset=${receiveAssetId}` +
+      `&destination=${receiveAssetAddress}`
+
+    const { data } = await axios.get<LendingDepositQuoteResponse>(url)
+    if (!data || 'error' in data)
+      return Err('Error fetching Thorchain lending deposit quote: no data received')
+
+    return Ok(data)
+  } catch (error) {
+    console.error('Error fetching Thorchain lending deposit quote:', error)
+    return Err(
+      `Error fetching Thorchain lending deposit quote: ${
+        error instanceof Error ? error.message : error
+      }`,
+    )
+  }
+}
+
+export const getMaybeThorchainLendingCloseQuote = async ({
+  repaymentAssetId,
+  repaymentAmountBaseUnit,
+  collateralAssetId,
+  loanOwner,
+}: {
+  repaymentAssetId: AssetId
+  repaymentAmountBaseUnit: BigNumber.Value | null | undefined
+  collateralAssetId: AssetId
+  loanOwner: string
+}): Promise<Result<LendingWithdrawQuoteResponse, string>> => {
+  if (!repaymentAmountBaseUnit) {
+    return Err('Repayment amount is required')
+  }
+
+  const repaymentAsset = selectAssetById(store.getState(), repaymentAssetId)
+  if (!repaymentAsset) return Err(`Asset not found for assetId ${repaymentAssetId}`)
+  const from_asset = assetIdToPoolAssetId({ assetId: repaymentAssetId })
+  if (!from_asset) return Err(`Pool asset not found for assetId ${repaymentAssetId}`)
+  const to_asset = assetIdToPoolAssetId({ assetId: collateralAssetId })
+  if (!to_asset) return Err(`Pool asset not found for assetId ${collateralAssetId}`)
+
+  const amount = toThorBaseUnit({
+    valueCryptoBaseUnit: repaymentAmountBaseUnit,
+    asset: repaymentAsset,
+  })
+
+  const url =
+    `${getConfig().REACT_APP_THORCHAIN_NODE_URL}/lcd/thorchain/quote/loan/close` +
+    `?from_asset=${from_asset}` +
+    `&amount=${amount.toString()}` +
+    `&to_asset=${to_asset}` +
+    `&loan_owner=${loanOwner}`
+
+  try {
+    const { data: quoteData } = await axios.get<LendingWithdrawQuoteResponse>(url)
+    if (!quoteData) {
+      return Err('No data received from THORChain')
+    }
+    if ('error' in quoteData) {
+      return Err(quoteData.error)
+    }
+    return Ok(quoteData)
+  } catch (error) {
+    return Err(`Error fetching THORChain lending withdraw quote: ${error}`)
+  }
+}
+
+export const getAllThorchainLendingPositions = async (
+  assetId: AssetId,
+): Promise<BorrowersResponseSuccess> => {
+  const poolAssetId = assetIdToPoolAssetId({ assetId })
+
+  if (!poolAssetId) throw new Error(`Pool asset not found for assetId ${assetId}`)
+
+  const { data } = await axios.get<BorrowersResponse>(
+    `${getConfig().REACT_APP_THORCHAIN_NODE_URL}/lcd/thorchain/pool/${poolAssetId}/borrowers`,
+  )
+
+  if (!data || 'error' in data) return []
+
+  return data
+}
+
+export const getThorchainLendingPosition = async ({
```

Comment:
can just leverage this endpoint to filter for us:
`/lcd/thorchain/pool/{pool}/borrower/{address}`

Reply:
Same reason we can't for savers - for UTXOs, we would need to make one request per address

Reply:
ah right :brain: